src/components/common/form/InputWithAdornment.js

```diff
@@ -0,0 +1,111 @@
+// @flow
+import React, { useCallback, useEffect } from 'react'
+import { isMobile, isMobileSafari } from 'mobile-device-detect'
+import { StyleSheet, TextInput, TouchableOpacity, View } from 'react-native'
+import normalize from '../../../lib/utils/normalizeText'
+import SimpleStore from '../../../lib/undux/SimpleStore'
+import { withStyles } from '../../../lib/styles'
+import Icon from '../view/Icon'
+import Config from '../../../config/config'
+import ErrorText from './ErrorText'
+
+const InputText = ({
+  adornment,
+  adornmentAction,
+  adornmentSize = 16,
+  adornmentStyle,
+  error,
+  styles,
+  theme,
+  style,
+  getRef,
+  ...props
+}) => {
+  const simpleStore = SimpleStore.useStore()
+  const shouldChangeSizeOnKeyboardShown = isMobileSafari && simpleStore.set && Config.safariMobileKeyboardGuidedSize
+
+  const onTouchStart = useCallback(() => {
+    if (shouldChangeSizeOnKeyboardShown) {
+      window.scrollTo(0, 0)
+      document.body.scrollTop = 0
+      simpleStore.set('isMobileSafariKeyboardShown')(true)
+    }
+
+    if (isMobile) {
+      simpleStore.set('isMobileKeyboardShown')(true)
+    }
+  }, [simpleStore])
+
+  const onBlur = useCallback(() => {
+    if (shouldChangeSizeOnKeyboardShown) {
+      simpleStore.set('isMobileSafariKeyboardShown')(false)
+    }
+
+    if (isMobile) {
+      simpleStore.set('isMobileKeyboardShown')(false)
+    }
+
+    if (props.onBlur) {
+      props.onBlur()
+    }
+  }, [simpleStore, props])
+
+  useEffect(() => {
+    simpleStore.set('isMobileSafariKeyboardShown')(false)
```

Comment:
Add this to `lib/undux/SimpleStore` & export:

```
const useCurriedSetters = (paths: string[]) => {
  const store = SimpleStore.useStore()

  return useMemo(
    () => paths.map(path => store.set(path)),
      [paths, store]
  )
}
```

Then you could write just:

```
const [setSafariKbdShown, setMobileKdbShown] = useCurriedSetters([
  'isMobileSafariKeyboardShown',
  'isMobileKeyboardShown'
])

// ...
setMobileKbdShown(true)
```

Reply:
done